extends StaticBody3D
class_name  Trash

var player : Player
@export var destino : Node3D
@export var offset : Vector3
var in_place : bool = false
@onready var carrying_detection: CarryingDetect = $Detection

func _ready() -> void:
	# Al iniciar, obtenemos las referencias de la baterÃ­a y el jugador
	player = get_tree().get_first_node_in_group("Player")
	carrying_detection.CanCarrying.connect(player_is_in_area)
	carrying_detection.NotCanCarrying.connect(player_is_not_in_area)
	
func _process(delta: float) -> void:
	var overlapping_bodies : Array[Node3D] = carrying_detection.get_overlapping_bodies()
	for body in overlapping_bodies:
		if body is Player:
			if player.can_take and Input.is_action_just_pressed("Take") and not in_place:
				reparent(player.carrying_offset)
				player.is_carrying = true
				global_position = player.carrying_offset.global_position + offset
				global_rotation = player.carrying_offset.global_rotation
				player.basura_tomar.play()
				player.carrying_object = self
				player.can_take = false
		
func player_is_in_area():
		player.is_in_area = true
		if not player.is_carrying and not in_place:
			player.can_take = true
func player_is_not_in_area():
	player.is_in_area = false
	player.can_take = false
